modules/Database/Arangojs.js

```diff
@@ -240,6 +240,85 @@ class ArangoJS {
         return result;
     }
 
+    async findLocalQuery(queryObject) {
+        const {
+            identifierKey,
+            opcode,
+            idType,
+        } = queryObject;
+
+        const queryParams = {
+            identifierKey,
+            idType,
+        };
+
+        let operator = '';
+        switch (opcode) {
+        case 'EQ':
+            operator = '==';
+            break;
+        case 'IN':
+            operator = 'IN';
+            break;
+        default:
+            throw new Error(`OPCODE ${opcode} is not supported`);
+        }
+
+        const queryString = `LET identifierObjects = (
+                                FOR v IN ot_vertices
+                                FILTER v.vertexType == "Identifier"
+                                AND v.identifierType == @idType
+                                and v.identifierValue ${operator} @identifierKey
+                                RETURN v
+                            )
+                            
+                            // Fetch the start entity for trail
+                            LET startObjects = UNIQUE(FLATTEN(
+                                FOR identifierObject IN identifierObjects
+                                    FILTER identifierObject != null
+                                    LET identifiedObject = (
+                                    FOR v, e IN 1..1 OUTBOUND identifierObject ot_edges
+                                    FILTER e.edgeType == 'IdentifierRelation'
+                                    RETURN v
+                                    )
+                                RETURN identifiedObject
+                            ))
+                            
+                            LET startObjectsWithTimestamps = (
+                            FOR startObject in startObjects
+                                FILTER startObject != null
+                                FOR dataset in startObject.datasets
+                                    FOR d IN ot_datasets
+                                        FILTER d._key == dataset
+                                        return {otObject: startObject, timestamp: d.datasetHeader.datasetCreationTimestamp})
```

Comment:
Because you have the exact key of the dataset, I suggest using `DOCUMENT` here for filtering the `ot_datasets` collection to avoid this loop

Reply:
There is `opcode` parameter which can't be used with `DOCUMENT` function